packages/block-editor/src/components/link-control/index.js

```diff
@@ -181,7 +161,10 @@ function LinkControl( {
 							key={ `${ suggestion.id }-${ suggestion.type }` }
 							itemProps={ buildSuggestionItemProps( suggestion, index ) }
 							suggestion={ suggestion }
-							onClick={ () => onLinkChange( suggestion ) }
+							onClick={ () => {
+								setIsEditingLink( false );
```

Comment:
Related to previous comment - it looks like we're now handling the switching of modes manually as required ratherr than relying on a `useEffect` dependency?